### ISSUE TITLE
Email plugin EmailEventListener('order-confirmation') missed some usefull information `shippingMethod.name, productVariant.featureAsset` ..

### ISSUE THREAD

**Issue by tianyingchun** [vendure-ecommerce/vendure#1354]
**Describe the bug**
for `plugin-email` we want to customized `EmailEventListener('order-confirmation')`

but it missed as below usefull information
1.  shippingMethod.name, description cause of it has type `LocaleString` need to join table.
```ts
 for (const line of context.event.order.shippingLines || []) {
            let shippingMethod: ShippingMethod | undefined;
            if (!line.shippingMethod && line.shippingMethodId) {
                shippingMethod = await context.injector
                    .get(TransactionalConnection)
                    .getRepository(ShippingMethod)
                    .findOne(line.shippingMethodId);
            } else if (line.shippingMethod) {
                shippingMethod = line.shippingMethod;
            }
            if (shippingMethod) {
                shippingMethods.push(shippingMethod);
            }
        }

```
2.  `.setTemplateVars(event => ({ order: event.order, shippingMethods: event.data.shippingMethods }))`
the `order.line.productVariant.featuredAsset` has missed also.

3.  `.loadData(async ({ event, injector }) => {`  should be also support `Language` configuration for diferent language locale.

**Expected behavior**
none

**Environment (please complete the following information):**
 - @vendure/core version: 1.4.4
 - Nodejs version 14.x+
 - Database (mysql/postgres etc): mysql

**Additional context**

**Comment by michaelbromley** [comment]
I've solved issue 1.

Regarding issue 2, we are currently referencing `OrderLine.featuredAsset`, which _is_ populated. The was an error with the correct asset path (which I have fixed) but other than that the data is there.

I'm not sure I fully understand point 3. It sounds like maybe a separate feature request. But maybe you can explain a bit more what you want to achieve here and then we decide.

**Comment by tianyingchun** [comment]
question 3: it should be ok, i see the event return me `ctx` properties , but orderline FeaturedAsset return me null

![image](https://user-images.githubusercontent.com/2833251/149856454-61a8c3af-0615-4dde-a207-5c9b1225d939.png)

**Comment by tianyingchun** [comment]
it seems it need specificed `product level ` featuredAsset :)

**Comment by tianyingchun** [comment]
BTW if we configuration asset-plugin as 
```ts
    AssetServerPlugin.init({
      route: 'assets',
      previewMaxHeight: 600,
      previewMaxWidth: 1200,
      assetUrlPrefix: 'https://dev.flatjs.com/assets/',
```

the `event.order.lines[0].featuredAsset` should return me full path http url but now only return me 

```ts
{
createdAt:Fri Dec 31 2021 22:41:42 GMT+0800 (China Standard Time)
fileSize:72367
focalPoint:null
height:600
id:7
  mimeType:'image/jpeg'
  name:'e110b-1__02.jpg'
  preview:'preview/1b/e110b-1__02__preview.jpg'
  source:'source/19/e110b-1__02.jpg'
  type:'IMAGE'
  updatedAt:Fri Dec 31 2021 22:41:42 GMT+0800 (China Standard Time)
  width:1200
}
```

**Comment by Tomaszal** [comment]
@michaelbromley the asset url fix seems to be missing from the current version (1.4.7) even though it was included in the changelog for version 1.4.5. Is there a reason it's excluded or was there a mistake?

**Comment by michaelbromley** [comment]
Hi @Tomaszal. This fix is due to be released in v1.5. I don't see it in the changelog - can you point out where it is listed?

**Comment by Tomaszal** [comment]
Oh sorry, I saw the line "email-plugin Fix population of shippingLines in order handler ([4cd7ecd](https://github.com/vendure-ecommerce/vendure/commit/4cd7ecd)), closes [#1354](https://github.com/vendure-ecommerce/vendure/issues/1354)" and assumed it included both of the fixes mentioned in this issue. But I see it's only the first one. Thanks for the info.

**Comment by michaelbromley** [comment]
Oh yeah, the system I use to generate the changelog always says "closes #1234" even if the commit only actually includes the text "Relates to #1234". I can see how the confusion came about.
